src/core_editor/editor.rs

```diff
@@ -181,10 +181,15 @@ impl Editor {
         self.line_buffer = val.clone();
     }
 
-    pub fn remember_undo_state(&mut self, is_after_action: bool) -> Option<()> {
```

Comment:
Having a function do two different things based on a boolean that is passed usually results in unreadable and
unmaintainable code and this is no different, in my opinion.

We reap the following advantages (if this is done)
- users of the interface don't have to worry about what that boolean flag is.
   - Given that Rust does not have kwargs the readability of passing booleans into a function results in having to read the 
      interface and documentation (or implementation) of the particular function. Which we can also avoid if we don't do it.
- removes the possibility of creating a bug because a wrong boolean value was selected.

Reply:
I think the already existing `UndoBehavior` enum would be a prime candidate to make that explicit and exhaustive (what `is_after_action` means is elusive to me now)